### PR TITLE
Component library with Backstage blog post

### PR DESCRIPTION
**Pull request by minkimcello**
WIP

### CODE REVIEW

File: src/blog/2022-02-20-component-library-backstage.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+If using the frontend component provided by Backstage plugins is less important to you than using your own component library, then there is something else you should know. In addition to Material UI components, the Backstage frontend framework provides a data fetching API for retrieving data from the backend. You can use your component library instead of Material UI but I would strongly recommend that you continue to use the data fetching Apis that come with Backstage plugins. This will allow you to still use Backstage plugins even if you’re not using the components that they provide.
+
+In this tutorial, I will show you how to replace Material UI with [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/controls/web) component library, then use Catalog Client API provided by the Catalog plugin to display a list of entities from the catalog in FluentUI’s DetailList. Let’s get started!
```
Comment (minkimcello) [inline src/blog/2022-02-20-component-library-backstage.md]: Elrick:
> Should we also mention that this was a specific use case from a client somewhere in the introduction as well? So people don't think we are going OffRoad here? I think that maybe a good addition to accompanying the great argument for not doing this approach. That will cover us as well if Spotify was to read this as well. :-)

Taras:
> I didn't add the bit about clients, but I think the updated version covers it pretty well. Give it a read and let me know what you think

**Comment by jorgelainfiesta** [comment]
@taras can I take this for editing? I'll inline a few code snippets from your links and will try to follow it to take the screenshots

**Comment by minkimcello** [comment]
@jorgelainfiesta can we send it on monday? i haven't made my suggestions yet

File: src/blog/2022-02-20-component-library-backstage.md (hunk 1 above)
Comment (minkimcello) [reply src/blog/2022-02-20-component-library-backstage.md]: @elrickvm The new intro can be found highlighted [here](https://github.com/thefrontside/frontside.com/pull/186/files#diff-fa5b287b0ab25697e359c410c88196303bcabc5564adff4c22a41d2a06e87e6dR14-R20)

File: src/blog/2022-02-20-component-library-backstage.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+export const apis: AnyApiFactory[] = [
++   createApiFactory({
++     api: catalogApiRef,
++     deps: { discoveryApi: discoveryApiRef },
++     factory: ({ discoveryApi }) => new CatalogClient({ discoveryApi })
++   }),
+  createApiFactory({
+```
+
+The catalog API reference comes from the `@backstage/plugin-catalog-react` package. This reference will be used to look up this API in your components when using the `useApi` hook. This factory controls how the Catalog Client is created. `CatalogClient` needs `DiscoveryApiRef` which provides the factory access to the `DiscoveryApi`. The DiscoveryApi is used to find the URL where the Backstage Catalog API can be found. This is the piping that makes it easy for us to call the API in future steps.
```
Comment (minkimcello) [inline src/blog/2022-02-20-component-library-backstage.md]: I think this paragraph could be rewritten. It's very difficult to follow.

**Review by minkimcello — COMMENTED** [review]
The netlify preview link to this post is [here](https://deploy-preview-186--frontside.netlify.app/blog/2022-02-20-component-library-backstage/).

I was originally planning on just converting the github commit links to code examples but ended up having to rewrite some of the paragraphs so that they're easier to understand. @taras could you read through and make sure the changes I made are still correct with its terminologies? You can see all the changes I made in [this](https://github.com/thefrontside/frontside.com/pull/186/commits/9b65260d36584bc66d24ecce5f1f784f94cd16b2) commit.

Also, @jorgelainfiesta I used `tsx` markdown code format for "new" files and `diff` format if we're _modifying_ a file so it's easier to keep track as the tutorial makes incremental changes to the same few files. However, the `diff` style of our website (see preview link) is not very pretty (whereas in vscode and github it's color-coded like [this](https://github.com/thefrontside/frontside.com/pull/186/commits/9b65260d36584bc66d24ecce5f1f784f94cd16b2#diff-fa5b287b0ab25697e359c410c88196303bcabc5564adff4c22a41d2a06e87e6dR99-R106)).

Let me know if I should change the examples from `diff` to `tsx` and I'll update them or if we can modify the styles of `diff` on our website.

**Comment by jorgelainfiesta** [comment]
@minkimcello thanks for this! I think the `diff` is better, I'll look into a way of incorporating it into our website. I also want to change the color highlighter, it's not very good so might as well find something that takes care of both 

@taras let me know if this is ready to be taken to Olson

**Comment by taras** [comment]
it's great. let's send it to olsen

**Review by jorgelainfiesta — APPROVED** [review]
Looks good to me! let's merge it on Monday

**Pull request merged**